RepoMan/RepoMan/Repository/BitBucketCloudPullRequestReader.cs

```diff
@@ -0,0 +1,359 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using RepoMan.Records;
+using ILogger = Serilog.ILogger;
+
+namespace RepoMan.Repository
+{
+    public class BitBucketCloudPullRequestReader :
+        IRepoPullRequestReader
+    {
+        private readonly string _prApiUrl;
+        private readonly string _repoTag;
+        private readonly HttpClient _bbClient;
+        private readonly JsonSerializerSettings _jsonSerializerSettings;
+        private readonly IClock _clock;
+        private readonly ILogger _logger;
+
+        public BitBucketCloudPullRequestReader(string hostname, string repoOwner, string repoName, HttpClient client, JsonSerializerSettings jsonSerializerSettings, IClock clock, ILogger logger)
+        {
+            if (!Uri.TryCreate(hostname, UriKind.Absolute, out var _)) throw new ArgumentException($"'{hostname}' is not a valid URL");
```

Comment:
Should we also check its a bitbucket hostname?

Reply:
I don't think it adds anything to check for it. For example, I don't know if it's on the BB Cloud roadmap to white label their offering with custom domains. Getting it wrong seems similar to getting an API key wrong.